### PR TITLE
AutoPR enhancements

### PR DESCRIPTION
**Pull request by mathias-baumann-sociomantic**
* D2 support
* Mention breaking changes
* Include links/commits in commit message
* Update, even if unsupported version

### CODE REVIEW

**Comment by mihails-strasuns-sociomantic** [comment]
Do you have an example of how mentioning breaking changes looks like?

**Comment by mathias-baumann-sociomantic** [comment]
It would look like this: 

This PR has been automatically created by *neptune-autopr* and
updates the following submodules:

* **makd** from version **v1.9.0** to version **v1.9.2** ([Release notes](https://github.com/sociomantic-tsunami/makd/releases/tag/v1.9.2), [Changed commits](https://github.com/sociomantic-tsunami/makd/compare/v1.9.0...v1.9.2))
  * Breaking Updates: [v1.2.3](https://github.com/sociomantic-tsunami/makd/releases/tag/v1.2.3), [v1.2.4](https://github.com/sociomantic-tsunami/makd/releases/tag/v1.2.4), [v1.2.5](https://github.com/sociomantic-tsunami/makd/releases/tag/v1.2.5)
* **ocean** from version **v3.3.1** to version **v3.3.5** ([Release notes](https://github.com/sociomantic-tsunami/ocean/releases/tag/v3.3.5), [Changed commits](https://github.com/sociomantic-tsunami/ocean/compare/v3.3.1...v3.3.5))
* **swarm** from version **v4.2.3** to version **v4.2.4** ([Release notes](https://github.com/sociomantic-tsunami/swarm/releases/tag/v4.2.4), [Changed commits](https://github.com/sociomantic-tsunami/swarm/compare/v4.2.3...v4.2.4))

**Comment by mathias-baumann-sociomantic** [comment]
Ping?

File: src/autopr/LibInfo.d [hunk 1]
```diff
@@ -270,22 +272,29 @@ struct LibInfo
     ***************************************************************************/
 
     bool markSupported ( bool complete, int maintained_minor_versions,
-        int maintained_major_months, Release[] releases )
+        int maintained_major_months, Release[] releases, bool d2_only )
```
Comment (mihails-strasuns-sociomantic) [inline src/autopr/LibInfo.d]: `Flag` or own enum please :angel:

File: src/autopr/LibInfo.d [hunk 2]
```diff
@@ -270,22 +272,29 @@ struct LibInfo
...
     {
         import std.range;
         import std.algorithm;
         import std.datetime;
 
-        auto onlyRCorFull ( Release rel )
+        auto d2filter ( Release rel )
+        {
+            return d2_only == rel.ver.metadata.canFind("d2");
```
Comment (mihails-strasuns-sociomantic) [inline src/autopr/LibInfo.d]: Will also match `v1.2.3+r2d2` :)

This highlights deficiency in our `Version` module btw - there can be several metadata blocks, it should really be parsed as an array of strings.

File: src/autopr/main.d [hunk 3]
```diff
@@ -600,14 +600,26 @@ void processSubmodules ( Json edge, LibInfo lib_info, RequestLevel global,
             continue;
         }
 
+        bool d2_only = false;
+
+        if (!meta_info.neptune_yaml.isNull() &&
+            meta_info.neptune_yaml.containsKey("d2ready") &&
+            meta_info.neptune_yaml["d2ready"] == "only")
+        {
+            d2_only = true;
```
Comment (mihails-strasuns-sociomantic) [inline src/autopr/main.d]: This does not support `.D2-ready` anchor file, right? So we need to ensure all repos move that info to neptune yml.

File: src/autopr/LibInfo.d (hunk 2 above)
Comment (mathias-baumann-sociomantic) [reply src/autopr/LibInfo.d]: It does match yes. `d2` is a substring of `r2d2` so `canFind` will discover it

File: src/autopr/main.d (hunk 3 above)
Comment (mathias-baumann-sociomantic) [reply src/autopr/main.d]: Correct, only the `.neptune.yml` content is considered

**Comment by mathias-baumann-sociomantic** [comment]
Updated

File: src/autopr/LibInfo.d (hunk 2 above)
Comment (leandro-lucarella-sociomantic) [reply src/autopr/LibInfo.d]: But this is not what we want, right? We want explicitly `+d2` I think.

**Comment by mathias-baumann-sociomantic** [comment]
Updated `Version` to provide metadata as array. so now we are explicitly searching for `d2` and would not match `r2d2`

**Pull request merged**